src/game/client/water/w3dwatertracks.cpp

```diff
@@ -325,9 +329,204 @@ int WaterTracksObj::Update(int ms_elapsed)
 
 void Test_Water_Update()
 {
-    // todo needs InGameUI
-#ifdef GAME_DLL
-    Call_Function<void>(PICK_ADDRESS(0x007A22C0, 0x0064DC2D));
+#ifdef BUILD_WITH_D3D8
+    static bool pause_waves;
+    static POINT mouse_anchor;
+    static Coord3D terrain_point_start;
+    static Coord3D terrain_point_end;
+    static WaterTracksObj *track;
+    static WaterTracksObj *track2;
+    static int track_edit_mode;
+    static int have_start;
+    static int have_end;
+    static int do_init = 1;
+    static WaveType current_wave_type = WAVE_TYPE_OCEAN;
+    static int track_edit_mode_reset = 1;
+    static int add_point_reset = 1;
+    static int delete_track_reset = 1;
+    static int save_tracks_reset = 1;
+    static int load_tracks_reset = 1;
+    static int change_type_reset = 1;
+
+    pause_waves = false;
+
+    if (do_init != 0) {
+        do_init = 0;
+    }
+
+    if ((GetAsyncKeyState(VK_F5) & 0x8001) != 0) {

```

Comment:
Could write as
```c++
enum : short
{
    TOGGLED = 0x0001,
    PRESSED = 0x8000,
}

if ((GetAsyncKeyState(VK_F5) & (TOGGLED | PRESSED)) != 0) {
```

Reply:
It's a win32 function i think. There should be constants for it already

Reply:
There are none. and the examples I find online (including directly from Microsoft) show no use of constants.

Reply:
Yes there are none. But we can define our own.

Reply:
Reading it more the only bits that change are MSB and LSB so this whole if can be simplified to `if(GetAsyncKeyState(VK_F5))`. LSB can't be relied upon either so perhaps it does make sense not to make an enum for a situation that doesn't really indicate anything useful.